libraries/oneview_helper.rb

```diff
@@ -18,31 +18,39 @@ module Helper
     # @param context Context from the resource action block (self)
     # @param [String, Symbol] type Name of the resource type. e.g., :EthernetNetwork
     # @param [String, Symbol] action Action method to call on the resource. e.g., :create_or_update
-    def self.do_resource_action(context, type, action)
-      klass = OneviewCookbook::Helper.get_resource_class(context, type)
+    # @param [Module] base_module The base API module supported. e.g. OneviewCookbook::ImageStreamer
+    def self.do_resource_action(context, type, action, base_module = OneviewCookbook)
+      klass = OneviewCookbook::Helper.get_resource_class(context, type, base_module)
       res = klass.new(context)
       res.send(action)
     end
 
     # Get the resource class by name, taking into consideration the resource's properties
     # @param context Context from the resource action block (self)
     # @param [String, Symbol] type Name of the resource type. e.g., :EthernetNetwork
+    # @param [Module] base_module The base API module supported. e.g. OneviewCookbook::ImageStreamer
     # @return [Class] Resource class
-    def self.get_resource_class(context, resource_type)
+    def self.get_resource_class(context, resource_type, base_module = OneviewCookbook)
       load_sdk(context)
-      api_version = context.property_is_set?(:api_version) ? context.api_version : context.node['oneview']['api_version']
-      api_module = get_api_module(api_version)
+      # Loads the api version if the property was specified directly
+      context_api_version = context.api_version if context.property_is_set?(:api_version)
+      # Loads the api version if it was specified in the client
+      client_api_version = convert_keys(context.client, :to_sym)[:api_version] if context.property_is_set?(:client)
+      # Loads the api version giving preference: property > client > node default
+      api_version = context_api_version || client_api_version || context.node['oneview']['api_version']
```

Comment:
@jsmartt  The way it was being defined before was completely ignoringthe Client API version.

Ideally, if no property is specified saying which API version should be used, the client API version must be used.

I think it was counterintuitive setting the client API version as 300, and then executing the API200, because it was the node default, but the preference was being overriden by the client.